core/sequencer/sequencer.go

```diff
@@ -262,47 +272,49 @@ func (s *Sequencer) applyMutations(mutations []*tpb.SignedKV, leaves []*trillian
 }
 
 // CreateEpoch signs the current map head.
-func (s *Sequencer) CreateEpoch(ctx context.Context, forceNewEpoch bool) error {
+func (s *Sequencer) CreateEpoch(ctx context.Context, forceNewEpoch bool) (*tpb.GetMutationsResponse, error) {
 	glog.V(2).Infof("CreateEpoch: starting sequencing run")
 	start := time.Now()
 	// Get the current root.
 	rootResp, err := s.tmap.GetSignedMapRoot(ctx, &trillian.GetSignedMapRootRequest{
 		MapId: s.mapID,
 	})
 	if err != nil {
-		return fmt.Errorf("GetSignedMapRoot(%v): %v", s.mapID, err)
+		return nil, fmt.Errorf("GetSignedMapRoot(%v): %v", s.mapID, err)
 	}
 	startSequence := rootResp.GetMapRoot().GetMetadata().GetHighestFullyCompletedSeq()
 	revision := rootResp.GetMapRoot().GetMapRevision()
 	glog.V(3).Infof("CreateEpoch: Previous SignedMapRoot: {Revision: %v, HighestFullyCompletedSeq: %v}", revision, startSequence)
 
 	// Get the list of new mutations to process.
-	mutations, seq, err := s.newMutations(ctx, startSequence)
+	mRange, seq, err := s.newMutations(ctx, startSequence)
 	if err != nil {
-		return fmt.Errorf("newMutations(%v): %v", startSequence, err)
+		return nil, fmt.Errorf("newMutations(%v): %v", startSequence, err)
 	}
 
 	// Don't create epoch if there is nothing to process unless explicitly
 	// specified by caller
-	if len(mutations) == 0 && !forceNewEpoch {
+	if len(mRange) == 0 && !forceNewEpoch {
 		glog.Infof("CreateEpoch: No mutations found. Exiting.")
-		return nil
+		return nil, nil
 	}
 
 	// Get current leaf values.
-	indexes := make([][]byte, 0, len(mutations))
-	for _, m := range mutations {
-		indexes = append(indexes, m.KeyValue.Key)
+	mutations := make([]*tpb.Mutation, 0, len(mRange))
+	indexes := make([][]byte, 0, len(mRange))
+	for _, m := range mRange {
+		mutations = append(mutations, &tpb.Mutation{Update: m})
```

Comment:
`tpb.Mutation` is a bit of a misnomer.  I'm about to rename it `tpb.MutationProof` since it is the combination of the mutation and its proof.  This might clarify some variable names here.  `mutations` was good the way it was since it is actually a list of mutations. The new `mutations` variable might be better named `mutationProofs` or something like that.